pkg/controller/controller.go

```diff
@@ -863,11 +873,64 @@ func convertAndFilterCatalogToNamespacedTypes(namespace string, in *osb.CatalogR
 	return serviceClasses, servicePlans, nil
 }
 
+// generateEscapedName takes in an OSB ID and filters
+// it to fit K8S name restrictions. It escapes all
+// characters except for lowercase alphanumerics (excluding z
+// as we use that for the escape character), hyphens, and
+// periods, changing them into their UTF-8 hex
+// equivalents separated by 'z'. Hyphens that come after periods and
+// periods that come after hyphens or periods are also escaped.
+// If the resulting string is over the K8S limit of 63 characters,
+// it appends the md5 hash of the original OSB ID to the first 31 characters
+// of the escaped name, with trailing periods trimmed.
+func generateEscapedName(externalID string) string {
+	buffer := bytes.Buffer{}
+	lenOrigin := len(externalID)
+	prevDot := false
+	prevDash := false
+
+	for i, ch := range externalID {
+		// Don't need to escape [a-z0-9.]
+		// '.' isn't ok at start, end or after '-', so escape those cases
+		if (ch >= 'a' && ch <= 'y') || (ch >= '0' && ch <= '9') {
+
+			buffer.WriteString(string(ch))
+			prevDash = false
+			prevDot = false
+		} else if ch == '.' && i != 0 && i != lenOrigin-1 && !prevDot && !prevDash {
+			buffer.WriteString(string(ch))
+			prevDash = false
+			prevDot = true
+		} else if ch == '-' && i != 0 && i != lenOrigin-1 && !prevDot {
+			buffer.WriteString(string(ch))
+			prevDash = true
+			prevDot = false
+		} else {
+			start, end := "z", "z" // By default start/end escaped ch with 'z'
+
+			buffer.WriteString(fmt.Sprintf("%s%x%s", start, ch, end)) // append
```

Comment:
this isn't different from any other writestring calls, so I don't think we need the append comment.